test/e2e/models/CodeServer.ts

```diff
@@ -282,6 +282,32 @@ export class CodeServerPage {
     this.codeServer.logger.debug("Editor is ready!")
   }
 
+  /**
+   * Checks if the test extension has loaded
+   *
+   * Reload until both checks pass
+   */
+  async reloadUntilTestExtensionIsLoaded() {
+    this.codeServer.logger.debug("Waiting for test extenion to load...")
+
+    const extensionIsLoaded = await this.isTestExtensionLoaded()
+    let reloadCount = 0
+
+    while (!extensionIsLoaded) {
+      await this.page.waitForLoadState("load")
+      // Give it an extra second just in case it's feeling extra slow
+      await this.page.waitForTimeout(1000)
+      reloadCount += 1
+      if (await this.isTestExtensionLoaded()) {
+        this.codeServer.logger.debug(`test extension loaded after ${reloadCount} reloads`)
```

Comment:
Tiny nit but I recommend using `plural` so we can get `1 reload` vs `2 reloads` etc.